samples/common/src/utils.rs

```diff
@@ -74,29 +71,34 @@ where
 /// Use Ibeji to discover the endpoint for a digital twin provider that satifies the requirements.
 ///
 /// # Arguments
-/// * `invehicle_digitial_twin_servuce_url` - In-vehicle digital twin service URL.
+/// * `invehicle_digitial_twin_servuce_uri` - In-vehicle digital twin service URI.
 /// * `entity_id` - The matching entity id.
 /// * `protocol` - The required protocol.
 /// * `operations` - The required operations.
 pub async fn discover_digital_twin_provider_using_ibeji(
-    invehicle_digitial_twin_servuce_url: &str,
+    invehicle_digitial_twin_servuce_uri: &str,
     entity_id: &str,
     protocol: &str,
     operations: &[String],
 ) -> Result<EndpointInfo, String> {
-    info!("Sending a find_by_id request for entity id {entity_id} to the In-Vehicle Digital Twin Service URL {invehicle_digitial_twin_servuce_url}");
-    let mut client = DigitalTwinClient::connect(invehicle_digitial_twin_servuce_url.to_string())
-        .await
-        .map_err(|error| format!("{error}"))?;
+    info!("Sending a find_by_id request for entity id {entity_id} to the In-Vehicle Digital Twin Service URI {invehicle_digitial_twin_servuce_uri}");
+    let mut client =
+        InvehicleDigitalTwinClient::connect(invehicle_digitial_twin_servuce_uri.to_string())
+            .await
+            .map_err(|error| format!("{error}"))?;
     let request = tonic::Request::new(FindByIdRequest { id: entity_id.to_string() });
     let response = client.find_by_id(request).await.map_err(|error| error.to_string())?;
     let response_inner = response.into_inner();
     debug!("Received the response for the find_by_id request");
     info!("response_payload: {:?}", response_inner.entity_access_info);
 
-    let entity_access_info = response_inner.entity_access_info.expect("Did not find the entity");
+    let entity_access_info = response_inner.entity_access_info;
```

Comment:
You can combine lines 95-104 using `ok_or()` and `?` and avoid the if statement:

```rust
match response_inner
    .entity_access_info
    .ok_or("error message")?
    .endpoint_info_list
    // etc...
```

Reply:
Fixed.